pinot-query-planner/src/main/java/org/apache/pinot/query/validate/Validator.java

```diff
@@ -147,4 +151,20 @@ protected void addToSelectList(List<SqlNode> list, Set<String> aliases,
     }
     super.addToSelectList(list, aliases, fieldList, exp, scope, includeSystemVars);
   }
+
+  @Override
+  public void validateWith(SqlWith with, SqlValidatorScope scope) {
+    Set<String> withNames = new HashSet<>();
+
+    for (SqlNode sqlWithItem : with.withList) {
+      for (String name : ((SqlWithItem) sqlWithItem).name.names) {
+        if (withNames.contains(name)) {
+          throw new RuntimeException("Duplicate alias in WITH: '" + name + "'");
```

Comment:
Can we also provide lines? Given we are at SqlNode here, we should be able to access them